### DISCUSSION TITLE
create a paragraph style with shadings

### DISCUSSION THREAD

**Discussion by toramameseven** [dolanmiu/docx#2314] [Q&A] [answered]
I want to create a paragraph style with shadings.
But, I can not set shading properties to paragraph.
Please, someone tell me how to set shading properties.

The sample code is next.

```sample.ts
import * as fs from "fs";
import { Document, Packer, Paragraph, ShadingType } from "docx";

const doc = new Document({
    styles: {
        paragraphStyles: [
            {
                id: "code",
                name: "code",
                run: {
                    color:"ff0000",
                    shading: {
                        type: ShadingType.PERCENT_15,
                        fill: "eeeeee",
                        color: "eeeeee"
                    },
                },
            },
        ],
    },
    sections: [
        {
            children: [
                new Paragraph({
                    text: "Hello World for not entire paragraph",
                    style: "code",
                }),
            ],
        },
    ],
});

Packer.toBuffer(doc).then((buffer) => {
    fs.writeFileSync("My Document.docx", buffer);
});
```

And the result is next.

The shading property is set to the font area not to the paragraph area.
And I can not edit the font area shading properties at Word for my skill.

![image](https://github.com/dolanmiu/docx/assets/72330367/e79947ce-9319-4600-a216-a7368af70923)

![image](https://github.com/dolanmiu/docx/assets/72330367/35fa1d47-41e1-4ee4-80a3-87c27082829a)

#531 is similar, but not for style.

**Comment by dolanmiu** [comment, marked as answer]
This is now done:

https://github.com/dolanmiu/docx/pull/669
